### PR TITLE
Login

### PR DESCRIPTION
**Pull request by ridhoq**
werk in progress

### CODE REVIEW

File: static/components/login/login.js [hunk 1]
```diff
@@ -0,0 +1,191 @@
+import React, {Component} from "react";
+import validator from "validator";
+import classNames from "classnames";
+import update from 'immutability-helper';;
```
Comment (austinpray) [inline static/components/login/login.js:4]: Rid fix your SHIT

File: static/components/login/login.js [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+    constructor(props) {
+        super(props);
+        this.state = {
+            user: {}
+        };
+    }
+
+    handleSubmit(e) {
+        e.preventDefault();
+        console.log("handled submit");
```
Comment (austinpray) [inline static/components/login/login.js:16]: Delet

File: static/components/login/login.js [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+            this.setState({
+                validEmail: false,
+                user: update(this.state.user, {email: {$set: null}})
+            });
+        }
+    };
+
+    handlePasswordChange = (e) => {
+        const val = e.target.value;
+        if (!validator.isEmpty(val) && validator.isAscii(val) && val.length >= 6) {
```
Comment (austinpray) [inline static/components/login/login.js:85]: You specify that the pw length must be 6 but never say it on UI

File: static/components/login/login.js [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+    }
+
+    handleSubmit(e) {
+        e.preventDefault();
+        console.log("handled submit");
+    };
+
+    render() {
+
+        // there's gotta be a better way to do this...
```
Comment (austinpray) [inline static/components/login/login.js]: Yeah create a wrapper component you dolt

Comment (austinpray) [reply static/components/login/login.js]: this.props.children

File: static/components/login/login.js [hunk 5]
```diff
@@ -0,0 +1,191 @@
...
+            "has-danger": this.state.validEmail !== null && this.state.validEmail === false
+        });
+
+        const passwordClass = classNames({
+            "form-control": true,
+            "form-control-success": this.state.user.password,
+            "form-control-danger": this.state.validPassword !== null && this.state.validPassword === false
+        });
+
+        const passwordGroup = classNames({
```
Comment (austinpray) [inline static/components/login/login.js]: Holy moly what's wrong with you

File: static/components/login/login.js [hunk 6]
```diff
@@ -0,0 +1,191 @@
...
+            "has-danger": this.state.validPassword !== null && this.state.validPassword === false
+        });
+
+        const confirmPasswordClass = classNames({
+            "form-control": true,
+            "form-control-success": this.state.validConfirmPassword,
+            "form-control-danger": this.state.validConfirmPassword !== null && this.state.validConfirmPassword === false
+        });
+
+        const confirmPasswordGroup = classNames({
```
Comment (austinpray) [inline static/components/login/login.js]: It keeps going

File: static/containers/songs/single.js [hunk 7]
```diff
@@ -55,7 +55,7 @@ export default class SongSingleContainer extends Component {
         }
 
         return <div>
-            <pre>{JSON.stringify(song, null, "  ")}</pre>
+            <p>{JSON.stringify(song, null, "  ")}</p>
```
Comment (austinpray) [inline static/containers/songs/single.js:58]: Nice change lol

**Review by austinpray — CHANGES_REQUESTED** [review]

File: static/components/login/login.js (hunk 3 above)
Comment (ridhoq) [reply static/components/login/login.js:85]: but I do tho, check the password input

**Pull request merged**